internal/watch/forwarders.go

```diff
@@ -85,8 +86,15 @@ func (ff Forwarders) DeleteAll() {
 // Kill stops and delete a port-forwards associated with pod.
 func (ff Forwarders) Kill(path string) int {
 	var stats int
+
+	// The way port forwards are stored is `pod_fqn|container|local_port:container_port`
+	// The '|' is added to make sure we do not delete port forwards from other pods that have the same prefix
+	// Without the `|` port forwards for pods, default/web-0 and default/web-0-bla would be both deleted
+	// even if we want only port forwards for default/web-0 to be deleted
+	prefix := fmt.Sprintf("%s|", path)
+
 	for k, f := range ff {
-		if strings.HasPrefix(k, path) {
+		if k == path || strings.HasPrefix(k, prefix) {
```

Comment:
I think just checking for the path is ok here. We don't need to add a prefix here since we are checking for `starts with` so if the match is partial or full this should satisfy the condition. Does this make sense?

Reply:
@derailed, no. the issue was that there was only `k == path`
in the `v.0.28.0` this was fixed, as you can see [here](https://github.com/derailed/k9s/commit/d19c792feb75aa576f05cdd0824470941cdc8bd4), but another bug is introduced: because you do not append the `|` (as per my comment above) you can delete more port forwards than desired. this is why I am stricter  with the check

Reply:
@placintaalexandru I stand corrected, you are absolutely right! Thank you!!